src/java.base/share/classes/jdk/internal/access/JavaLangReflectAccess.java

```diff
@@ -29,73 +29,43 @@
-    internals of java.lang.reflect. */
-
+    internals of java.lang.reflect. Use as a last resort! */
 public interface JavaLangReflectAccess {
-    /** Creates a new java.lang.reflect.Constructor. Access checks as
-      per java.lang.reflect.AccessibleObject are not overridden. */
-    public <T> Constructor<T> newConstructor(Class<T> declaringClass,
-                                             Class<?>[] parameterTypes,
-                                             Class<?>[] checkedExceptions,
-                                             int modifiers,
-                                             int slot,
-                                             String signature,
-                                             byte[] annotations,
-                                             byte[] parameterAnnotations);
-
-    /** Gets the MethodAccessor object for a java.lang.reflect.Method */
-    public MethodAccessor getMethodAccessor(Method m);
-
-    /** Sets the MethodAccessor object for a java.lang.reflect.Method */
-    public void setMethodAccessor(Method m, MethodAccessor accessor);
-
-    /** Gets the ConstructorAccessor object for a
-        java.lang.reflect.Constructor */
-    public ConstructorAccessor getConstructorAccessor(Constructor<?> c);
-
-    /** Sets the ConstructorAccessor object for a
-        java.lang.reflect.Constructor */
-    public void setConstructorAccessor(Constructor<?> c,
-                                       ConstructorAccessor accessor);
+    /**
+     * Creates a new root constructor from the original one, with
+     * a custom accessor. Used by serialization hooks.
+     */
+    <T> Constructor<T> newConstructorWithAccessor(Constructor<T> original, ConstructorAccessor accessor);
 
     /** Gets the byte[] that encodes TypeAnnotations on an Executable. */
     public byte[] getExecutableTypeAnnotationBytes(Executable ex);
 
-    /** Gets the "slot" field from a Constructor (used for serialization) */
-    public int getConstructorSlot(Constructor<?> c);
-
-    /** Gets the "signature" field from a Constructor (used for serialization) */
-    public String getConstructorSignature(Constructor<?> c);
-
-    /** Gets the "annotations" field from a Constructor (used for serialization) */
-    public byte[] getConstructorAnnotations(Constructor<?> c);
-
-    /** Gets the "parameterAnnotations" field from a Constructor (used for serialization) */
-    public byte[] getConstructorParameterAnnotations(Constructor<?> c);
+    //<editor-fold defaultstate="collapsed" desc="Trusted array sharing">
 
     /** Gets the shared array of parameter types of an Executable. */
     public Class<?>[] getExecutableSharedParameterTypes(Executable ex);
 
     /** Gets the shared array of exception types of an Executable. */
     public Class<?>[] getExecutableSharedExceptionTypes(Executable ex);
 
-    //
+    //</editor-fold>
+
+    //<editor-fold defaultstate="collapsed" desc="AccessibleObject Copying">
```

Comment:
Personal editor comments should not be included.

Reply:
Ah, thought editor folds were acceptable as they appear in a few places in javac code. Will remove.